File: Selenium/Activities/Popup.py
from selenium import webdriver
from selenium.webdriver.common.action_chains import ActionChains
from  selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.service import Service as ChromeService
from selenium.webdriver.support.select import Select
from webdriver_manager.chrome import ChromeDriverManager
driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()))

# Navigate to the URL
driver.get("https://www.v1.training-support.net/selenium/popups")
# Print the title of the page
print("Page title is: ", driver.title)

# Find the "Sign In" button
signin_button = driver.find_element(By.CLASS_NAME, "orange")
# Move the cursor over the button to show tooltip
ActionChains(driver).move_to_element(signin_button).perform()
# Print the tooltip text
tooltip_text = signin_button.get_attribute("data-tooltip")
print("Button tooltip text: ", tooltip_text)
# Click the button
signin_button.click()

# Find the username and password fields
username = driver.find_element(By.ID, "username")
password = driver.find_element(By.ID, "password")
# Enter credentials
username.send_keys("admin")
password.send_keys("password")
# Click login
driver.find_element(By.XPATH, "/html/body/div[4]/div/div[2]/div/div/button").click()

# Print the confirmation message
message = driver.find_element(By.ID, "action-confirmation")
print("Confirmation message: ", message.text)

File: Selenium/Activities/PopUp.java
package activities;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
        import org.openqa.selenium.WebDriver;
        import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
        import org.openqa.selenium.firefox.FirefoxOptions;
        import org.openqa.selenium.interactions.Actions;

public class PopUp {
    public static void main(String[] args) {
        WebDriverManager.chromedriver().setup();
        ChromeDriver driver= new ChromeDriver();

        // Open the webpage
        driver.get("https://www.v1.training-support.net/selenium/popups");

        // Print the title of the page
        String pageTitle = driver.getTitle();
        System.out.println("Page Title: " + pageTitle);

        // Find the Sign in button and hover over it
        WebElement signInButton = driver.findElement(By.id("signin"));
        Actions actions = new Actions(driver);
        actions.moveToElement(signInButton).perform();

        // Print the tooltip message
        WebElement tooltip = driver.findElement(By.className("tooltiptext"));
        String tooltipMessage = tooltip.getText();
        System.out.println("Tooltip Message: " + tooltipMessage);

        // Click the button to open the Sign in form
        signInButton.click();

        // Enter the credentials
        WebElement usernameField = driver.findElement(By.id("username"));
        WebElement passwordField = driver.findElement(By.id("password"));
        usernameField.sendKeys("admin");
        passwordField.sendKeys("password");

        // Click login
        WebElement loginButton = driver.findElement(By.xpath("//button[contains(text(),'Log in')]"));
        loginButton.click();

        // Print the message after logging in
        WebElement successMessage = driver.findElement(By.id("action-confirmation"));
        String loginSuccessMessage = successMessage.getText();
        System.out.println("Login Success Message: " + loginSuccessMessage);

        // Close the browser
        driver.quit();
    }
}
